plugins/ptp_operator/metrics/manager.go

```diff
@@ -0,0 +1,266 @@
+		Version: "v1",
+		Values: []ceevent.DataValue{{
+			Resource:  string(p.publisherTypes[eventType].Resource),
+			DataType:  ceevent.NOTIFICATION,
+			ValueType: ceevent.ENUMERATION,
+			Value:     state,
+		}, {
+			Resource:  string(p.publisherTypes[eventType].Resource),
+			DataType:  ceevent.METRIC,
+			ValueType: ceevent.DECIMAL,
+			Value:     float64(ptpOffset),
+		},
+		},
+	}
+	p.publish(data, source, eventType)
+}
+
+func (p *PTPEventManager) publish(data ceevent.Data, resource string, eventType ptp.EventType) {
+	if pubs, ok := p.publisherTypes[eventType]; ok {
+		e, err := common.CreateEvent(pubs.PubID, string(eventType), resource, data)
+		if err != nil {
+			log.Errorf("failed to create ptp event, %s", err)
+			return
+		}
+		if !p.mock {
+			if err = common.PublishEventViaAPI(p.scConfig, e); err != nil {
+				log.Errorf("failed to publish ptp event %v, %s", e, err)
+				return
+			}
+		}
+	} else {
+		log.Errorf("failed to publish ptp event due to missing publisher for type %s", string(eventType))
+	}
+}
+
+// GenPTPEvent ... generate events form the logs
+func (p *PTPEventManager) GenPTPEvent(ptpProfileName string, stats *stats.Stats, eventResourceName string, ptpOffset int64, clockState ptp.SyncState, eventType ptp.EventType) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("Recovered in f", r)
+		}
+	}()
+	if clockState == "" {
+		return
+	}
+
+	lastClockState := stats.LastSyncState()
+	threshold := p.PtpThreshold(ptpProfileName)
+	switch clockState {
+	case ptp.LOCKED:
+		switch lastClockState {
+		case ptp.FREERUN: //last state was already sent for FreeRUN , but if its within then send again with new state
+			if isOffsetInRange(ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold) { // within range
+				log.Infof(" publishing event for ( profile %s) %s with last state %s and current clock state %s and offset %d for ( Max/Min Threshold %d/%d )",
+					ptpProfileName, eventResourceName, stats.LastSyncState(), clockState, ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold)
+				p.PublishEvent(clockState, ptpOffset, eventResourceName, eventType) // change to locked
+				stats.SetLastSyncState(clockState)
+				stats.SetLastOffset(int64(ptpOffset))
+				stats.AddValue(int64(ptpOffset)) // update off set when its in locked state and hold over only
+			}
+		case ptp.LOCKED: // last state was in sync , check if it is out of sync now
+			if isOffsetInRange(ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold) {
+				stats.SetLastOffset(int64(ptpOffset))
+				stats.AddValue(int64(ptpOffset)) // update off set when its in locked state and hold over only
+			} else {
+				log.Infof(" publishing event for ( profile %s) %s with last state %s and current clock state %s and offset %d for ( Max/Min Threshold %d/%d )",
+					ptpProfileName, eventResourceName, stats.LastSyncState(), clockState, ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold)
```

Comment:
same comment here:  use `lastClockState` from L208.

Reply:
fixed

Reply:
I don't see it fixed